src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import javax.naming.AuthenticationException;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User userFromDB = userService.findByEmail(email).orElseThrow(() ->
+                new AuthenticationException("Incorrect username or password!"));
```

Comment:
let's make user service return user or null

Reply:
Why does it need to return the user or null? In my opinion, it's better to leave an exception

Reply:
our service should not return optional.So here we will check if(user != null && passwordsDoesn'tMatch) then throw new  Exception